packages/components/namespace-store/reducer.tsx

```diff
@@ -1,37 +1,48 @@
 import { AWS_REGIONS } from '../../constants';
 
-export type StoreAction =
+export type NSSFormAction =
   | { type: 'setSecretName'; value: string }
   | { type: 'setSecretKey'; value: string }
   | { type: 'setAccessKey'; value: string }
   | { type: 'setRegion'; value: string }
   | { type: 'setTarget'; value: string }
-  | { type: 'setEndpoint'; value: string };
+  | { type: 'setEndpoint'; value: string }
+  | { type: 'setName'; value: string }
+  | { type: 'setPVC'; value: string }
+  | { type: 'setPVCFolderName'; value: string };
```

Comment:
create enum for ActionType and use that here as well as while dispatching any state...